### PR TITLE
core-lib web app

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: website_react/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "short_name": "React App",
+  "name": "Create React App Sample",
+  "icons": [
+    {
+      "src": "favicon.ico",
+      "sizes": "64x64 32x32 24x24 16x16",
+      "type": "image/x-icon"
+    },
+    {
+      "src": "logo192.png",
```
Comment (shay-te) [inline website_react/public/manifest.json:11]: we can use the corelib logo now

File: website_react/src/components/dataCollectionForms/dataAccessForm.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import InputString from '../inputs/InputString'
```
Comment (shay-te) [inline website_react/src/components/dataCollectionForms/dataAccessForm.jsx:1]: Filename will be same as class name will start with capital D

File: website_react/src/App.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				default_value: 'UserCoreLib',
+				mandatory: true,
+				validatorCallback: validateFunc,
+			},
+			// DB CONNECTION
+			
+
+			// // DB Entity
+			// {
+			// 	title: 'Enter the number of columns',
```
Comment (shay-te) [inline website_react/src/App.js:24]: add another property called value for showing a form that was already filled before

File: website_react/src/components/dataCollectionForms/dataAccessForm.jsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+import InputInteger from '../inputs/InputInteger'
+import InputBoolean from '../inputs/InputBoolean'
+import InputENUM from '../inputs/InputENUM'
+import InputList from '../inputs/InputList'
+
+
+const DataAccessForm = () => {
```
Comment (shay-te) [inline website_react/src/components/dataCollectionForms/dataAccessForm.jsx:8]: In this implementation, we are letting the View (React) manage our data.
we can hold this fields array for each class we are falling and reuse a single component called `Form`
also a code duplication `DataAccessForm`, `DBConnectionForm`, `DBEntityForm`, and so on and so fourth 
What do you thnk about crating a single component called `Form` and reuse it with different data each time?

File: website_react/src/components/dataCollectionForms/dbConnectionForm.jsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                title: "Enter your DB password",
+                type: "string",
+                default_value: null,
+                mandatory: true,
+                validatorCallback: validateFunc,
+            },
+        ],
+    };
+	return (
+		<>
```
Comment (shay-te) [inline website_react/src/components/dataCollectionForms/dbConnectionForm.jsx:98]: code duplication. 
you could to here `return (<Form fields={fields}/>)`.

but just create the JSON lists upfront. and we will pass them each time to a Form component

File: website_react/src/components/form/form.scss [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    background-color: rgb(105, 255, 205);
+    border: none;
+    padding: 0.5rem;
+    font-size: 1rem;
+    border-radius: 5px;
+    border: 1px black solid;
+}
+
+/* Extra small devices (phones, 600px and down) */
+@media only screen and (max-width: 600px) {
```
Comment (shay-te) [inline website_react/src/components/form/form.scss:44]: let's remove ALL media queries for now. for the reason that we don't have a design yet. or at least decided on one.

also not sure we need them

File: website_react/src/components/inputs/InputBoolean.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    return (
+        <div className="form-input-div" key={props.index}>
+            <label className="input-label">{props.formFields.title}</label>
+            <div>
+                <input
+                    type="radio"
+                    id={"true" + props.index}
+                    name={"bool" + props.index}
+                    value="true"
+                    defaultChecked={props.formFields.default_value}
```
Comment (shay-te) [inline website_react/src/components/inputs/InputBoolean.jsx:11]: what happens id default_value is undefined?

File: website_react/src/components/inputs/InputBoolean.jsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+const InputBoolean = (props) => {
+    return (
+        <div className="form-input-div" key={props.index}>
+            <label className="input-label">{props.formFields.title}</label>
+            <div>
+                <input
+                    type="radio"
+                    id={"true" + props.index}
```
Comment (shay-te) [inline website_react/src/components/inputs/InputBoolean.jsx:8]: are we/the form is using the id?

File: website_react/src/components/inputs/InputBoolean.jsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                    name={"bool" + props.index}
+                    value="true"
+                    defaultChecked={props.formFields.default_value}
+                    required={props.formFields.mandatory}
+                />
+                <label htmlFor={"true" + props.index}>True</label>
```
Comment (shay-te) [inline website_react/src/components/inputs/InputBoolean.jsx:14]: yes

File: website_react/src/components/inputs/InputBoolean.jsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+            <div>
+                <input
+                    type="radio"
+                    id={"false" + props.index}
+                    name={"bool" + props.index}
+                    value="false"
+                    defaultChecked={!props.formFields.default_value}
+                    required={props.formFields.mandatory}
+                />
+                <label htmlFor={"false" + props.index}>False</label>
```
Comment (shay-te) [inline website_react/src/components/inputs/InputBoolean.jsx:26]: no

File: website_react/src/components/inputs/InputBoolean.jsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
+const InputBoolean = (props) => {
+    return (
+        <div className="form-input-div" key={props.index}>
+            <label className="input-label">{props.formFields.title}</label>
+            <div>
+                <input
```
Comment (shay-te) [inline website_react/src/components/inputs/InputBoolean.jsx:6]: are these two inputs are grouped?

File: website_react/src/components/inputs/InputENUM.jsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
+const InputENUM = (props) => {
+    return (
+        <div className="form-input-div" key={props.index}>
+            <label className="input-label">{props.formFields.title}</label>
+            {props.formFields.options.map((value, index) => {
```
Comment (shay-te) [inline website_react/src/components/inputs/InputENUM.jsx:5]: 1 rule for react components (my rule)
No logic. No loops. No ifs. inside the JSX.  
suggestions 
```
const renderOption = (option) = > ...return jsx here 

const options = props.options.map(renderOption)

    return (
         <div className="form-input-div" key={props.index}>
             <label className="input-label">{props.formFields.title}</label>
             {options}.
           </div>)
```

File: website_react/src/components/inputs/InputList.jsx [hunk 13]
```diff
@@ -0,0 +1,29 @@
+const InputList = (props) => {
+    return (
+        <div className="form-input-div" key={props.index}>
+            <label className="input-label">{props.formFields.title}</label>
+            {props.formFields.options.map((value, index) => {
```
Comment (shay-te) [inline website_react/src/components/inputs/InputList.jsx:5]: same

File: website_react/src/components/inputs/InputBoolean.jsx (hunk 7 above)
Comment (shubham-shay-te) [reply website_react/src/components/inputs/InputBoolean.jsx:11]: no field will be selected by default

File: website_react/src/components/inputs/InputBoolean.jsx (hunk 8 above)
Comment (shubham-shay-te) [reply website_react/src/components/inputs/InputBoolean.jsx:8]: no, but it's just a good practice to add `id` to form inputs

File: website_react/src/components/inputs/InputBoolean.jsx (hunk 11 above)
Comment (shubham-shay-te) [reply website_react/src/components/inputs/InputBoolean.jsx:6]: yes, as we will have to show both of them everytime for yes/no inputs

File: website_react/public/manifest.json (hunk 1 above)
Comment (shay-te) [reply website_react/public/manifest.json:11]: in your email

File: website_react/src/App.js (hunk 3 above)
Comment (shay-te) [reply website_react/src/App.js:24]: That is correct
`default_value` will  be there if no value exists. 
`value` is used when the user selected a value in advanced

File: website_react/src/components/dataCollectionForms/dataAccessForm.jsx (hunk 4 above)
Comment (shay-te) [reply website_react/src/components/dataCollectionForms/dataAccessForm.jsx:8]: let's talk about it.

File: website_react/src/components/dataCollectionForms/dbConnectionForm.jsx (hunk 5 above)
Comment (shay-te) [reply website_react/src/components/dataCollectionForms/dbConnectionForm.jsx:98]: if we have 
1. list of fields 
2. Form component that can render any form that we like 
do we need to add more component to render a new form?

File: website_react/src/components/inputs/InputBoolean.jsx (hunk 10 above)
Comment (shay-te) [reply website_react/src/components/inputs/InputBoolean.jsx:26]: what is the reason that this comment is resolved?

File: website_react/src/components/inputs/InputBoolean.jsx (hunk 9 above)
Comment (shay-te) [reply website_react/src/components/inputs/InputBoolean.jsx:14]: what was resolved?

**Pull request merged**